Component/Mass/src/sofa/component/mass/UniformMass.inl

```diff
@@ -179,7 +179,28 @@ void UniformMass<DataTypes>::initDefaultImpl()
         msg_info() << "Topology path used: '" << l_topology.getLinkedPath() << "'";
 
         d_indices.createTopologyHandler(meshTopology);
-        d_indices.addTopologyEventCallBack(sofa::core::topology::TopologyChangeType::ENDING_EVENT, [this](const core::topology::TopologyChange* eventTopo) {
+        /// Computations to be done when points are added
+        d_indices.addTopologyEventCallBack(sofa::core::topology::TopologyChangeType::POINTSADDED, [this](const core::topology::TopologyChange* eventTopo) {
+            WriteAccessor<Data<SetIndexArray > > indices = d_indices;
+            const core::topology::PointsAdded* pointsAdded = static_cast<const core::topology::PointsAdded*>(eventTopo);
+            auto& addedIndices = pointsAdded->getIndexArray();
+
+            for (unsigned i = 0; i<addedIndices.size(); i++)
+            {
+                indices.push_back(addedIndices[i]);
+            }
```

Comment:
d_indices is already a subsetIndices, its resize on adding/removing event is already internally handled before calling the callback. So you are duplicating the work.
I'm quite surprised no scene-tests are crashing. Perhaps because all tests on topological changes are using Diagonal or MeshMatrixMass.
Normally UniformMass behavior was already fixe din PR: #2377
Do you confirm you experienced a bug of mass value? 

Looking at the issue pointed in the comment, it seems the scene is using only a MechanicalObject + ParticleSource (which create new dofs). There is no topology in the node so all those callback are not created.

Reply:
I can confirm the bug (as I was with @hugtalbot for the investigation 🕵️‍♂️), and the test scene, SPHParticleSource.scn does have a PointSetTopologyContainer (and a modifier)